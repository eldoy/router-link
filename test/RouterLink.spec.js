import React from 'react'
import { Router } from '../index.js'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

describe('RouterLink', () => {
  it('should match the routes', () => {
    const routes = [
      { path: '/', view: 'Home' },
      { path: '/about', view: 'About' },
      { path: '/login', view: 'Login' },
      { path: '/chat', view: 'Chat' },
      { path: '*', view: 'NotFound' }
    ]

    const component = renderer.create(
      <Router routes={ routes }/>
    )

    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()

    function ss (path) {
      window.history.pushState({}, '', path)
      window.dispatchEvent(new Event('pushstate'))
      tree = component.toJSON()
      expect(tree).toMatchSnapshot()
    }
    ss('/about')
    ss('/login')
    ss('/chat')
    ss('/awefoijo23j')
  })
})
