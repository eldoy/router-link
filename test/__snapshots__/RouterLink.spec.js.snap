// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouterLink should match the routes 1`] = `
<Home
  route={Object {}}
/>
`;

exports[`RouterLink should match the routes 2`] = `
<About
  route={Object {}}
/>
`;

exports[`RouterLink should match the routes 3`] = `
<Login
  route={Object {}}
/>
`;

exports[`RouterLink should match the routes 4`] = `
<Chat
  route={Object {}}
/>
`;

exports[`RouterLink should match the routes 5`] = `
<NotFound
  route={Object {}}
/>
`;
